### PR TITLE
task-8 done

### PR DESCRIPTION
**Pull request by KitTheProgrammer**

### CODE REVIEW

File: src/api/index.ts [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
-        dispatch(setFilms(formattedData))
-    }
-}
\ No newline at end of file
+  const res = await getMethod(`movies${(params) ? `?${getParamsString(params)}` : ''}`)
+  if (res.status === 200) {
+    const data: { data: MovieDataFromServer[] } = await res.json()
+    const formattedData: FilmData[] = data.data.map((m) => movieToFilm(m))
+    dispatch(setFilms(formattedData))
+  }
```
Comment (Twixs) [inline src/api/index.ts]: please add error handling and show appropriate message to the user. do the same for all requests

File: src/components/FormInput/index.tsx [hunk 2]
```diff
@@ -1,34 +1,61 @@
...
+          value={value}
+          onChange={({ target: { value } }) => setValue(value)}
+        />
     }
   }
 
   return <div className={`main-form-input ${className || ''}`}>
     <span>{label}</span>
+    {meta.error && meta.touched && <div>{meta.error}</div>}
     {getInput()}
```
Comment (Twixs) [inline src/components/FormInput/index.tsx]: pass type to `getInput` directly, don't relay on closure

File: src/redux/reducers/films.ts [hunk 3]
```diff
@@ -31,10 +31,37 @@ export const filmsSlice = createSlice({
     },
     setSelectedSortOption: (state, action: PayloadAction<number>) => {
       state.selectedSortOption = action.payload
+    },
+    updateFilm: (state, action: PayloadAction<FilmData>) => {
+      const { payload } = action
+      const filmsCopy = [...state.films]
+      const filmInd = filmsCopy.findIndex((it) => it.id === payload.id)
+      if (filmInd !== -1) {
+        filmsCopy.splice(filmInd, 1, payload)
```
Comment (Twixs) [inline src/redux/reducers/films.ts]: it's better to use direct insert by id `filmsCopy[filmInd] = payload` for better readability and micro optimization

File: src/api/index.ts [hunk 4]
```diff
@@ -1,22 +1,62 @@
...
-            description: m.overview,
-            rating: m.vote_average,
-            runtime: m.runtime,
-        }))
-        dispatch(setFilms(formattedData))
+      const data: { data: MovieDataFromServer[] } = await res.json()
+      const formattedData: FilmData[] = data.data.map((m) => movieToFilm(m))
+      dispatch(setFilms(formattedData))
+    } else {
+      dispatch(setError({ error: true, errorMessage: 'Error at fetching the movies (server error)' }))
```
Comment (Twixs) [inline src/api/index.ts:14]: try catch is enough. no need for "else"

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            "windows"
+        ],
+        "quotes": [
+            "error",
+            "single"
+        ],
+        "semi": [
+            "error",
+            "never"
+        ]
```
Comment (Twixs) [inline .eslintrc.json:39]: please always use semicolon. it's a good practice

Comment (KitTheProgrammer) [reply .eslintrc.json:39]: no

**Pull request merged**